From 30fbdcf8d58658860e43c0396cb50c5d4ade2dc5 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Date: Sat, 4 Jul 2020 03:04:38 +0300
Subject: [PATCH] dt/dtoverlays: imx219: Add parameter to set camera
 module rotation

Add a rotation DT overlay parameter to allow specifying the camera
module mounting rotation. Set the default rotation to 180 as the module
is typically mounted upside-down.

Signed-off-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
---
 arch/arm/boot/dts/overlays/README             | 5 +++--
 arch/arm/boot/dts/overlays/imx219-overlay.dts | 6 ++++++
 2 files changed, 9 insertions(+), 2 deletions(-)

--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -1433,8 +1433,9 @@ Name:   imx219
 Info:   Sony IMX219 camera module.
         Uses Unicam 1, which is the standard camera connector on most Pi
         variants.
-Load:   dtoverlay=imx219
-Params: <None>
+Load:   dtoverlay=imx219,<param>=<val>
+Params: rotation                Mounting rotation of the camera sensor (0 or
+                                180, default 180)
 
 
 Name:   imx290
--- a/arch/arm/boot/dts/overlays/imx219-overlay.dts
+++ b/arch/arm/boot/dts/overlays/imx219-overlay.dts
@@ -27,6 +27,8 @@
 				VDIG-supply = <&imx219_vdig>;	/* 1.8v */
 				VDDL-supply = <&imx219_vddl>;	/* 1.2v */
 
+				rotation = <180>;
+
 				port {
 					imx219_0: endpoint {
 						remote-endpoint = <&csi1_ep>;
@@ -110,4 +112,8 @@
 			cam0-pwdn      = <&imx219_vana>,"gpio:4";
 		};
 	};
+
+	__overrides__ {
+		rotation = <&imx219>,"rotation:0";
+	};
 };
